### ISSUE TITLE
GNOME Settings no longer shows the Remote Desktop settings

### ISSUE THREAD

**Issue by michaelgrahamevans** [NixOS/nixpkgs#361163]
## Describe the bug

GNOME Settings does not show the Remote Desktop row and settings in the System page.

## Steps To Reproduce

1. Install GNOME
1. Enable the `gnome-remote-desktop` service with `services.gnome.gnome-remote-desktop.enable = true;`
2. Open GNOME Settings
3. Go to the System page

<details>
<summary>Here's a flake I used for testing</summary>

```nix
{
  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-24.11";

  outputs = { self, nixpkgs }: {
    nixosConfigurations = {
      test-gnome-remote-desktop = nixpkgs.lib.nixosSystem {
        system = "x86_64-linux";
        modules = [
          {
            services.xserver = {
              enable = true;
              desktopManager.gnome.enable = true;
              displayManager.gdm.enable = true;
            };

            services.gnome.gnome-remote-desktop.enable = true;

            users.users.nixos = {
              initialPassword = "password";
              isNormalUser = true;
              group = "nixos";
            };
            users.groups.nixos = { };
          }
        ];
      };
    };
  };
}
```

`nixos-rebuild build-vm --flake .#test-gnome-remote-desktop`

</details>

## Expected behavior

The System page should show a Remote Desktop row and clicking the row should allow me to edit settings and enable remote desktop connections.

### Remote Desktop row showing in GNOME 46 / NixOS 24.05

![Remote Desktop row showing in GNOME 46 / NixOS 24.05](https://github.com/user-attachments/assets/44462490-5c79-49be-b775-da1b00ab6b96)

### Remote Desktop row _not_ showing in GNOME 47 / NixOS 24.11

![Remote Desktop row _not_ showing in GNOME 47 / NixOS 24.11](https://github.com/user-attachments/assets/e79c12dc-9146-4b8c-a23b-564f9a07f807)

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 24.11 (Vicuna), 24.11.20241130.62c435d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/ljk3wvwqqzbmz0pys49ii18qmsi5aibc-source`

## Notify maintainers

@NixOS/gnome 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jtojnar** [comment]
Thanks for reporting. It appears is a bug in GNOME Settings.

Whether the [button is visible](https://gitlab.gnome.org/GNOME/gnome-control-center/-/blob/fd62b9016f0e9f2b5ae8a4affd65dbea088d7d6d/panels/system/cc-system-panel.c#L168-171) is controlled by the state of the [`gnome-remote-desktop.service`](https://gitlab.gnome.org/GNOME/gnome-control-center/-/blob/fd62b9016f0e9f2b5ae8a4affd65dbea088d7d6d/panels/system/cc-systemd-service.h#L24) system systemd service. Now that calls a [D-Bus API](https://gitlab.gnome.org/GNOME/gnome-control-center/-/blob/fd62b9016f0e9f2b5ae8a4affd65dbea088d7d6d/panels/system/cc-systemd-service.c#L41-70) that converts the variants into enum value. Unfortunately, the set of supported variants is [incomplete](https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.systemd1.html#Properties1) and does not contain `linked`, which is returned on NixOS:


```ShellSession
$ gdbus call --system \
                                                   --dest org.freedesktop.systemd1 \
                                                   --object-path /org/freedesktop/systemd1 \
                                                   --method org.freedesktop.systemd1.Manager.GetUnitFileState \
                                                   "gnome-remote-desktop.service"
('linked',)
```

I have opened an issue upstream: https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/3281

In the meanwhile, you should be able to set it up using `grdctl` CLI program.

**Comment by nadrog** [comment]
> In the meanwhile, you should be able to set it up using `grdctl` CLI program.

If by "using grdctl", one should use `sudo grdctl --system rdp enable`, then this can't work as it would need to change readonly files. I get the following:

```
Could not enable system service:
Failed to enable gnome-remote-desktop.service: GDBus.Error:System.Error.EROFS: File /etc/systemd/system/graphical.target.wants/gnome-remote-desktop.service: Read-only file system
```

Am I doing something wrong?

**Comment by jtojnar** [comment]
Yeah, imperative modification of system-wide systemd services will not work on NixOS. In this case, the following NixOS option might help.

```nix
    systemd.services.gnome-remote-desktop = {
      wantedBy = [ "graphical.target" ];
    };
```

**Comment by nadrog** [comment]
Thanks, this works!

**Comment by ghost** [comment]
It works. But we still need fix. Thanks!

**Comment by sbpro86** [comment]
> Yeah, imperative modification of system-wide systemd services will not work on NixOS. In this case, the following NixOS option might help.
> 
>     systemd.services.gnome-remote-desktop = {
>       wantedBy = [ "graphical.target" ];
>     };

@jtojnar 

The **desktop sharing** mode is working but not the **Remote login** and the `gnome-remote-desktop` service is stopping when the user is logged out while it should be active for the **remote login** mode.

Anyway thanks for the trick.

**Comment by stefankreutz** [comment]
I am unable to set up remote *login* (headless multi-user) but remote *assistance* (connecting to a runnnig GNOME session) works as described in [gnome-remote-desktop's README](https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/blob/0040410bfaf197eeb43d02b641bdaab3253994ac/README.md) with current nixos-unstable:

Log in via SSH, and forward the RDP port:

```bash
ssh -t -o "ExitOnForwardFailure yes" -L 3389:localhost:3389 example.com -- systemd-run --user --wait --pty -- bash -il
```

Create a self-signed TLS certificate:

```bash
mkdir -p ~/.local/share/gnome-remote-desktop/
openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout ~/.local/share/gnome-remote-desktop/tls.key -out ~/.local/share/gnome-remote-desktop/tls.crt -subj '/CN=localhost'
```

Configure gnome-remote-desktop service:

```bash
grdctl rdp set-tls-key ~/.local/share/gnome-remote-desktop/tls.key
grdctl rdp set-tls-cert ~/.local/share/gnome-remote-desktop/tls.crt                                                                                                                                                                            grdctl rdp clear-credentials
grdctl rdp set-credentials # May trigger graphical keychain creation prompt.
grdctl rdp set-port 3389 # Default RDP port.
grdctl rdp disable-port-negotiation
grdctl rdp disable-view-only                                                                                                                                                                                                                   grdctl rdp enable
```

Enable gnome-remote-desktop-service declaratively:

```nix
systemd.user.services.gnome-remote-desktop = {
  wantedBy = [ "default.target" ];
  unitConfig.ConditionUser = "!@system";
};
```

... or imperatively:

```bash
systemctl --user enable --now gnome-remote-desktop.service # Unless enabled declaratively.
```

Connect using an RDP client:

```bash
wlfreerdp /v:localhost /cert:ignore /u:alex /p:sesame
```
